File: measure-android/measure/src/main/java/sh/measure/android/config/ScreenshotMaskLevel.kt
package sh.measure.android.config

/**
 * The level of masking to apply to the screenshot.
 */
enum class ScreenshotMaskLevel {
    /**
     * The strictest level of masking which masks all text, input fields, images and videos.
     */
    AllTextAndMedia,

    /**
     * Masks all text and input fields, including clickable elements.
     */
    AllText,

    /**
     * Masks all text and input fields, excluding clickable elements.
     */
    AllTextExceptClickable,

    /**
     * The most lenient level of masking which only masks sensitive input fields like passwords,
     * email and phone number fields.
     */
    @Suppress("unused")
    SensitiveFieldsOnly,
}

File: flutter/packages/measure_flutter/lib/src/config/screenshot_mask_level.dart
import 'package:json_annotation/json_annotation.dart';

enum ScreenshotMaskLevel {
  /// Masks all text, input fields, images and videos.
  @JsonValue('all_text_and_media')
  allTextAndMedia,

  /// Masks all text and input fields, including clickable elements.
  @JsonValue('all_text')
  allText,

  /// Masks all text and input fields, excluding clickable elements.
  @JsonValue('all_text_except_clickable')
  allTextExceptClickable,

  /// Masks only sensitive input fields like passwords, email and phone fields.
  @JsonValue('sensitive_fields_only')
  sensitiveFieldsOnly,
}
